compiler/ras/DebugCounter.cpp

```diff
@@ -413,8 +420,15 @@ TR::DebugCounter *TR::DebugCounterGroup::createCounter(const char *name, int8_t
       flags |= TR::DebugCounter::ContributesToDenominator;
       }
    TR::DebugCounter *result = new (persistentMemory) TR::DebugCounter(name, fidelity, denominator, flags);
+   TR_ASSERT(result, "DebugCounter *result must not be null. Ensure availability of persistent memory");
```

Comment:
Without asserts, we would get a segmentation fault in the next few lines when we access the result pointer. In testing, this only happens when persistentMemory has run out of space, because of a massive number of debug counters. Example: On 31-bit z/OS, this will trigger after about 1.4 million debug counters were created on some personal build tests with `debugCounters={*}`. In practice, this is rarely likely to occur, since developers would only enable the few debug counters they want to see. The idea of capping the number of debug counters based on a proportion of memory available was brought up, but after discussing with @fjeremic, choosing a good number would not be trivial due to the different environments/configurations available.